gazebo_plugins/worlds/gazebo_ros_ackermann_drive_demo.world

```diff
@@ -0,0 +1,660 @@
+<?xml version="1.0"?>
+<!--
+  Gazebo ROS ackermann drive plugin demo
+
+  Try sending commands:
+
+    ros2 topic pub /demo/cmd_demo geometry_msgs/Twist '{linear: {x: 5.0}}' -1
+
+    ros2 topic pub /demo/cmd_demo geometry_msgs/Twist '{angular: {z: 0.1}}' -1
+
+  Try listening to odometry:
+
+    ros2 topic echo /demo/odom_demo
+
+  Try listening to TF:
+
+    ros2 run tf2_ros tf2_echo odom_demo chassis
+
+    and other links
+-->
+<sdf version="1.6">
+  <world name="default">
+    <include>
+      <uri>model://sun</uri>
+    </include>
+
+    <include>
+      <uri>model://ground_plane</uri>
+    </include>
+
+    <model name='vehicle'>
```

Comment:
Right now the model is slightly different from the one on the database. How about including the model from the database instead of describing a new model to avoid duplication? Something like this:

```
    <include>
      <uri>model://prius_hybrid</uri>
      <plugin ....>
        ...
       </plugin>
    </include>
```

Reply:
The model in the database does not have all the joints and links required. I got this model from [osrf/car_demo](https://github.com/osrf/car_demo/blob/master/prius_description/urdf/prius.urdf). I removed the sensors and changed the mesh files to load from those in `prius_hybrid` which the database contains.

Reply:
Oh interesting, the models on the database use universal joints on their front axle. On [citysim](https://bitbucket.org/osrf/citysim/src/d258e487ca560159a3b592f77642e5bd125969c0/worlds/simple_city.world#lines-14077) we used `libPriusHybridPlugin` with the models from the database and used the same joint for `front_left_wheel` and `front_left_wheel_steering`. I honestly don't know how that was working though :thinking:

We can leave this the way it is since it works, but now I'm just curious about citysim.

Reply:
I tried `ackermann_drive` the same way (specifying same joint for `front_left` and `front_left_steering`) and it works. Also the wrong TF problem and the vibration in `demo/distance_demo` are fixed. I think we will use this then

Reply:
I just noticed these lines are wrong:

https://github.com/ros-simulation/gazebo_ros_pkgs/blob/b3426432d64322f57c6b233a5dd6741c1ccb71f0/gazebo_plugins/src/gazebo_ros_ackermann_drive.cpp#L471-L472

The argument passed to `Position()` is the axis index. But no joints have 3 axes, so an index of 2 is invalid. The reason why nothing breaks when using a revolute joint is that the [index is ignored](https://bitbucket.org/osrf/gazebo/src/9e18a31cda6328c1d8d4ec08b9d1b6cfcdb93585/gazebo/physics/ode/ODEHingeJoint.cc#lines-128). But for a universal joint, an index of `2` should [print an error](https://bitbucket.org/osrf/gazebo/src/9e18a31cda6328c1d8d4ec08b9d1b6cfcdb93585/gazebo/physics/ode/ODEUniversalJoint.cc#lines-145).

If we want to support both revolute and universal joints, we could use an index `1`, which should work at least for ODE. I haven't checked the other engines.

Reply:
Yes,  I had to change that to index 0 to make it work.

Reply:
But I am having aome other problems with the odometry with the database model.  I guess model's X is not the car's front.  When the car moves,  the odom moves in Y direction

Reply:
Ah interesting, the database model is facing -Y for some reason. We don't need to change the model if it will be too much work. It would be document these requirements that we're finding though. That the joints need to be revolute with one axis, the model must be facing +X, etc.

Reply:
I have used the database model in the latest commit as it was more stable wrt odometry and had correct TF. I had to change the model to rotate it, so could not avoid duplication of the model.